tests/unit_tests/dft/include/compute_out_of_place.hpp

```diff
@@ -23,73 +23,79 @@
+    auto [forward_distance, backward_distance] =
+        get_default_distances<domain>(sizes, strides_fwd, strides_bwd);
+    auto ref_distance = std::accumulate(sizes.begin(), sizes.end(), 1, std::multiplies<>());
 
     descriptor_t descriptor{ sizes };
     descriptor.set_value(oneapi::mkl::dft::config_param::PLACEMENT,
                          oneapi::mkl::dft::config_value::NOT_INPLACE);
     descriptor.set_value(oneapi::mkl::dft::config_param::NUMBER_OF_TRANSFORMS, batches);
-    descriptor.set_value(oneapi::mkl::dft::config_param::FWD_DISTANCE, forward_elements);
+    descriptor.set_value(oneapi::mkl::dft::config_param::FWD_DISTANCE, forward_distance);
     descriptor.set_value(oneapi::mkl::dft::config_param::BWD_DISTANCE, backward_distance);
-    if constexpr (domain == oneapi::mkl::dft::domain::REAL) {
+    if (strides_fwd.size()) {
+        descriptor.set_value(oneapi::mkl::dft::config_param::INPUT_STRIDES, strides_fwd.data());
+    }
+    if (strides_bwd.size()) {
+        descriptor.set_value(oneapi::mkl::dft::config_param::OUTPUT_STRIDES, strides_bwd.data());
+    }
+    else if constexpr (domain == oneapi::mkl::dft::domain::REAL) {
         const auto complex_strides = get_conjugate_even_complex_strides(sizes);
         descriptor.set_value(oneapi::mkl::dft::config_param::OUTPUT_STRIDES,
                              complex_strides.data());
     }
     commit_descriptor(descriptor, sycl_queue);
+    std::vector<FwdInputType> fwd_data(
+        strided_copy(input, sizes, strides_fwd, batches, forward_distance));
 
-    std::vector<FwdInputType> fwd_data(input);
-
+    auto tmp = std::vector<FwdOutputType>(
+        cast_unsigned(backward_distance * batches + get_default(strides_bwd, 0, 0L)), 0);
     {
         sycl::buffer<FwdInputType, 1> fwd_buf{ fwd_data };
-        sycl::buffer<FwdOutputType, 1> bwd_buf{ sycl::range<1>(
-            cast_unsigned(backward_distance * batches)) };
+        sycl::buffer<FwdOutputType, 1> bwd_buf{ tmp };
 
         oneapi::mkl::dft::compute_forward<descriptor_t, FwdInputType, FwdOutputType>(
             descriptor, fwd_buf, bwd_buf);
 
         {
             auto acc_bwd = bwd_buf.template get_host_access();
             auto bwd_ptr = acc_bwd.get_pointer();
-            auto ref_iter = out_host_ref.begin();
-            const auto ref_row_stride = sizes.back();
-            const auto backward_row_stride = cast_unsigned(get_backward_row_size<domain>(sizes));
-            const auto backward_row_elements = cast_unsigned(get_backward_row_size<domain>(sizes));
-
-            while (ref_iter < out_host_ref.end()) {
-                EXPECT_TRUE(check_equal_vector(bwd_ptr, ref_iter, backward_row_elements,
-                                               abs_error_margin, rel_error_margin, std::cout));
-                bwd_ptr += backward_row_stride;
-                ref_iter += ref_row_stride;
+            for (int64_t i = 0; i < batches; i++) {
```

Comment:
```suggestion
            for (std::int64_t i = 0; i < batches; i++) {
```
and other instances in this file

Reply:
fixed